### PR TITLE
lmgtfy: Add lmgtfy plugin[gsoc]

### PR DESCRIPTION
**Pull request by meetmangukiya**
I have to escape the square brackets otherwise it is not transmitted well, and we won't get a markdown link in the chat. See https://github.com/errbotio/errbot/issues/1018

# TODO
- [ ] After this PR has merged open an issue to remove the escapes after the bug is fixed upstream.

### CODE REVIEW

File: plugins/lmgtfy.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+class Lmgtfy(BotPlugin):
+    """
+    For all those people who find it more convenient to bother you with their
+    question rather than search it for themselves.
+    """
+
+    @re_botcmd(pattern=r"lmgtfy\s+(.+)")
+    def lmgtfy(self, msg, match):
+        return "https://www.lmgtfy.com/?q=" + match.group(1)
```
Comment (meetmangukiya) [inline plugins/lmgtfy.py]: Add the complete message

File: plugins/lmgtfy.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class Lmgtfy(BotPlugin):
+    """
+    For all those people who find it more convenient to bother you with their
+    question rather than search it for themselves.
+    """
+
+    MSG = (
+        "Hey! You seem to have asked a question which could have been solved "
```
Comment (Udayan12167) [inline plugins/lmgtfy.py]: Single quotes again. :)

File: plugins/lmgtfy.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        "by other means easily.Click [this]({}) When you ask a question, be "
```
Comment (Udayan12167) [inline plugins/lmgtfy.py]: easily -> as well.

File: plugins/lmgtfy.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        "sure to try out a few things first - you're in a much better position "
+        "to help yourself then we are. Try googling, thinking, the GitHub "
```
Comment (Udayan12167) [inline plugins/lmgtfy.py]: then -> than

File: plugins/lmgtfy.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        "search or git grep if you are looking for source code."
```
Comment (Udayan12167) [inline plugins/lmgtfy.py]: the github search -> searching on github.

Comment (Udayan12167) [inline plugins/lmgtfy.py]: Let's link the git grep page here using markdown if possible.

File: plugins/lmgtfy.py (hunk 2 above)
Comment (meetmangukiya) [inline plugins/lmgtfy.py]: Okay, I'll configure QuotesBear to use single quotes then

**Comment by meetmangukiya** [comment]
@Udayan12167 I've applied QuotesBear and even made different directory for the plugin. Do you want to make different directories for plugins, it looks like a messy import after that. Maybe we just add plugins to the same directory for now and then if needed we move to directory structure?

**Comment by meetmangukiya** [comment]
Even the Quotesbear patch is different... some of them are double and some of them are single. It seems like QuoteBear is still buggy, do you want me to revert 51d79e4 ? @Udayan12167 ?

File: plugins/lmgtfy/lmgtfy.plug [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+name = lmgtfy
+
+[Documentation]
+description = For all those people who find it more convenient to bother you with their question rather than search it for themselves.
+
+[Python]
+version = 3
+
+[Errbot]
+
```
Comment (meetmangukiya) [inline plugins/lmgtfy/lmgtfy.plug]: remove this line

File: plugins/lmgtfy/lmgtfy.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+class Lmgtfy(BotPlugin):
+    """
+    For all those people who find it more convenient to bother you with their
+    question rather than search it for themselves.
+    """
+
+    MSG = (
+        'Hey! You seem to have asked a question which could have been solved '
+        'by other means as well.Click [this]({}) When you ask a question, be '
```
Comment (Udayan12167) [inline plugins/lmgtfy/lmgtfy.py]: Missing space after period `well. Click` also missing period after `this`

File: plugins/lmgtfy.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+class Lmgtfy(BotPlugin):
+    """
+    For all those people who find it more convenient to bother you with their
+    question rather than search it for themselves.
+    """
+
+    MSG = (
+        'Hey! You seem to have asked a question which could have been solved '
+        'by other means as well. Click [this]({}). When you ask a question, be '
+        "sure to try out a few things first - you're in a much better position "
```
Comment (Udayan12167) [inline plugins/lmgtfy.py]: double quotes

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/57/commits/6ce21e5a5d63c73fa342eac268ba28d10942fa56

**Comment by meetmangukiya** [comment]
@rultor merge

**Pull request merged**